app/controllers/course/assessment/submission/submissions_controller.rb

```diff
@@ -9,9 +9,11 @@ class Course::Assessment::Submission::SubmissionsController < \
   before_action :check_password, only: [:edit, :update]
   before_action :load_or_create_answers, only: [:edit, :update]
   before_action :check_zombie_jobs, only: [:edit]
+  before_action :create_submission_questions, only: :create
```

Comment:
When new questions are created, you also need to make sure new question_submissions are created. so they are also keep changing to some extent.